bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/ThingLinkyRemoteHandler.java

```diff
@@ -0,0 +1,880 @@
+        this.loadCurveConsumption = new ExpiringDayCache<>("loadCurveConsumption", REFRESH_HOUR_OF_DAY,
+                REFRESH_MINUTE_OF_DAY, () -> {
+                    LocalDate today = LocalDate.now();
+                    MeterReading meterReading = getLoadCurveConsumption(today.minusDays(6), today);
+                    meterReading = getMeterReadingAfterChecks(meterReading);
+                    if (meterReading != null) {
+                        logData(meterReading.baseValue, "Day (peak)", DateTimeFormatter.ISO_LOCAL_DATE, Target.ALL);
+                    }
+                    return meterReading;
+                });
+    }
+
+    @Override
+    public synchronized void initialize() {
+        logger.debug("Initializing Linky handler for {}", config.prmId);
+
+        // reread config to update timezone field
+        config = getConfigAs(LinkyThingRemoteConfiguration.class);
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            return;
+        }
+
+        if (bridge.getHandler() instanceof BridgeRemoteBaseHandler bridgeHandler) {
+            enedisApi = bridgeHandler.getEnedisApi();
+            divider = bridgeHandler.getDivider();
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            if (config.seemsValid()) {
+                if (config.timezone.isBlank()) {
+                    zoneId = this.timeZoneProvider.getTimeZone();
+                } else {
+                    zoneId = ZoneId.of(config.timezone);
+                }
+
+                bridgeHandler.registerNewPrmId(config.prmId);
+                pollingJob = scheduler.schedule(this::pollingCode, 5, TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.config-error-mandatory-settings");
+            }
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        super.updateStatus(status, statusDetail, description);
+    }
+
+    public boolean supportNewApiFormat() throws LinkyException {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            throw new LinkyException("Unable to get bridge in supportNewApiFormat()");
+        }
+
+        if (bridge.getHandler() instanceof BridgeRemoteBaseHandler bridgeHandler) {
+            return bridgeHandler.supportNewApiFormat();
+        } else {
+            throw new LinkyException("Unable to get bridgeHandler in supportNewApiFormat()");
+        }
+    }
+
+    private void pollingCode() {
+        try {
+            EnedisHttpApi api = this.enedisApi;
+
+            if (api != null) {
+                Bridge lcBridge = getBridge();
+                ScheduledFuture<?> lcPollingJob = pollingJob;
+
+                if (lcBridge == null || lcBridge.getStatus() != ThingStatus.ONLINE) {
+                    return;
+                }
```

Comment:
For your information, you are not handling properly bridge/thing but I will fix that in a next PR to not delay the merge.